app/src/main/java/com/yacgroup/yacguide/AboutActivity.kt

```diff
@@ -36,12 +38,33 @@ class AboutActivity : BaseNavigationActivity() {
         super.onCreate(savedInstanceState)
         this.title = getString(R.string.menu_about, getString(R.string.app_name))
 
-        val textViewAppVersion: TextView = findViewById(R.id.textViewAppVersion)
-        textViewAppVersion.text = "Version ${getAppVersion()}"
-        textViewAppVersion.gravity = Gravity.CENTER_HORIZONTAL
+        _createEntry(getString(R.string.data_src), getString(R.string.data_src_url),null)
+        _createEntry(getString(R.string.privacy_policy), null) { _showPrivacyPolicy() }
+        _createEntry(
+                getString(R.string.source_code_and_support),
+                getString(R.string.github_url),
+                null )
+        _createEntry(getString(R.string.license), getString(R.string.license_url),null)
+        _createEntry(getString(R.string.version), _getAppVersion(), null)
     }
 
-    private fun getAppVersion() : String {
+    private fun _createEntry(
+            title: String,
+            description: String?,
+            callback: ((view: View) -> Unit)?) {
+        val contentAbout: LinearLayout = findViewById(R.id.aboutContent)
+        val entry = layoutInflater.inflate(R.layout.about_entry, null)
+        if (callback == null && URLUtil.isValidUrl(description)) {
```

Comment:
I don't like having two parameters that contradict each other (`description` and `callback`). Too many permutations to be checked. 
1. You better enforce the callback parameter (non-null) and pass `_openUrl()` call to the method in case of `description` being a link. 
2. Since you use the value of `description` for every entry (by assignment to TextView), it should be non-null. Better use default-initialization.
```
_createEntry(titleA, { _showPrivacyBlaBlub() })
_createEntry(titleB, { _openUrl(desc) }, desc)
...

private fun _createEntry(title: String, callback: () -> Unit, description: String = "")
```
The check for valid url should be done in `_openUrl()`

Reply:
I understand your general point and fixed that. 

However, I don't like the solution to  pass the description twice. I found an alternative solution.